planet/clients/orders.py

```diff
@@ -98,21 +98,34 @@ def _stats_url(self):
     def _request(self, url, method, data=None, params=None, json=None):
         return Request(url, method=method, data=data, params=params, json=json)
 
-    async def _do_request(
-        self,
-        request: Request
-    ) -> Response:
+    def _order_factory(self, contents: dict) -> Order:
+        """Convert API response contents to an instance of Order."""
+        links = contents.get("_links", [])
+        results = []
+
+        for res in links.get("results", []):
+            expiration = res.get("expires_at", None)
+            if expiration:
+                expiration = arrow.get(expiration).datetime
+            results.append(
+                OrderComponent(delivery=res.get("delivery", None),
+                               name=res.get("name", None),
+                               location=res.get("location", None),
+                               expires_at=expiration))
+
+        return Order(id=contents["id"],
+                     state=contents["state"],
+                     results=results)
```

Comment:
@jreiberkyle here is the heart of this PR. Previously, the Order class constructor took a blob of JSON from the API response and did some computation to come up with an order description object. This coupled the API response directly to our information model. I have a hypothesis that using a less tightly coupled Order data class and shifting the responsibility for translation to the OrdersClient will give us more freedom and fun down the road.